calico/etcddriver/driver.py

```diff
@@ -457,6 +460,21 @@ def _handle_etcd_node(self, snap_mod, snap_key, snap_value,
         # several updates from the watcher queue (if there are
         # any).  We limit the number to ensure that we always
         # finish the snapshot eventually.
+        # PLW: what happens if we cannot keep up with the watcher Q, so it gets
+        # longer and longer forever? Should we have a maximum length?
+        # I think we ought to log the length of the queue when we come in here
+        # too, or at least have something so that we know what kind of queue
+        # lengths we have reached.
+        # PLW: This processes (normally) events off the queue (new changes)
+        # in preference to the current set of data. I think that is probably right,
+        # but for the initial load you could definitely argue that we should do
+        # something a bit different. For example, should we try to get to a consistent
+        # out of date state ASAP, rather than letting the new events come through?
+        # After some agonising, I concluded that letting new events win is right unless
+        # the new events are overwhelming the entire server, so this is probably correct.
+        # However, the number 100 then feels very arbitrary. Not sure that I have
```

Comment:
100 is fairly arbitrary but it's the right ballpark.  It's hard to hit >100 events for each parsed key so it's just a safety valve, really.